Weather/Weather/Controllers/WeatherCityController.swift

```diff
@@ -0,0 +1,134 @@
+    @IBOutlet weak var temperatureMinLabel: UILabel!
+    @IBOutlet weak var dayWeatherLabel: UILabel!
+    @IBOutlet weak var nightWeatherLabel: UILabel!
+    @IBOutlet weak var weatherFiveDays: UITableView!
+    @IBOutlet weak var navigationBar: UINavigationItem!
+    @IBOutlet weak var scrollRefresh: UIScrollView!
+    lazy var refreshControl: UIRefreshControl = {
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action:
+            #selector(handleRefresh(_:)),
+                                 for: UIControlEvents.valueChanged)
+        return refreshControl
+    }()
+    var model: WeatherModelProtocol!
+    var modelDelegate: CitiesModel!
+    var hud: MBProgressHUD?
+
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.scrollRefresh.addSubview(self.refreshControl)
+        navigationBar.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .trash, target: nil, action: #selector(deleteCity))
+        navigationBar.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: nil, action: #selector(addCity))
+        navigationBar.rightBarButtonItem?.tintColor = .red
+        if model.isLoad {
+            updateInterface()
+        } else if Reachability.isConnectedToNetwork() {
+            isHiddenView(true)
+            hud = MBProgressHUD.showAdded(to: self.view, animated: true)
+            hud?.mode = .indeterminate
+            hud?.detailsLabel.text = "Please wait"
+            model.updateData {
+                self.updateInterface()
+            }
+        } else {
+            self.showAlert(title: "error", message: "Internet connection")
+        }
+    }
+    
+    private func isHiddenView(_ value: Bool) {
+        for i in 0..<view.subviews.count {
+            view.subviews[i].isHidden = value
+        }
+    }
+    
+    @objc private func handleRefresh(_ refreshControl: UIRefreshControl) {
+        if Reachability.isConnectedToNetwork() {
+            model.updateData {
+                self.updateInterface()
+                refreshControl.endRefreshing()
+            }
+        } else {
+            self.showAlert(title: "error", message: "Internet connection")
+        }
+    }
+    
+    @objc func addCity() {
+        guard let controller = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: NewCityController.reuseIdentifier) as? NewCityController else { return }
+        controller.modelDelegate = modelDelegate
+        self.present(controller, animated: true, completion: nil)
+    }
+    
+    @objc func deleteCity() {
+        if !modelDelegate.deleteCity(city: model.city) {
+            showAlert(title: "Error", message: "You can not delete this page")
+        }
+    }
+    
+    func updateInterface() {
+        DispatchQueue.main.async {
+            self.cityLabel.text = self.model.city
+            if let maxTemp = self.model.weatherDay?.temperatureMax {
```

Comment:
What if data not exists (nil)? Will your app display trash or empty space? Both is bad